### DISCUSSION TITLE
ST7789V Driver

### DISCUSSION THREAD

**Discussion by suntchi2018** [Bodmer/TFT_eSPI#3011] [Q&A - General]
### Discussed in https://github.com/Bodmer/TFT_eSPI/discussions/2070

<div type='discussions-op-text'>

<sup>Originally posted by **shamirtahir50** October 10, 2022</sup>
Hi, 

I am using TFT024B045 which contains ST7789V, I am programmoing it using ESP32.

I am defining this in User_Setup.h

#define ESP32_PARALLEL

#define ST7789_DRIVER

// ESP32 pins used for the parallel interface TFT
#define TFT_CS   27  // Chip select control pin
#define TFT_DC   14  // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST  26  // Reset pin

#define TFT_WR   12  // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD   13

#define TFT_D0   16  // Must use pins in the range 0-31 for the data bus
#define TFT_D1   4  // so a single register write sets/clears all bits
#define TFT_D2   23
#define TFT_D3   22
#define TFT_D4   21
#define TFT_D5   19
#define TFT_D6   18
#define TFT_D7   17

Is this the right way?
I need to change the pin configuration or not?
Do you have any documentation about that?

</div>

**Comment by suntchi2018** [comment]
Hi to all,
I'm sorry but I'm a beginner in this environment, if I've made any mistakes I'm sorry, if you can show me the best way unfortunately I don't know how to create a new discussion.

I'm trying to get the TF 240x320 display to work with a ST7789V using a DOIT ESP32 DEVKIT V1 38 PIN with SPI, but the display behaves badly, the images loaded from the SD, don't have the correct colors, and some tests with other examples work with different colors or even appear in negative.

My configuration in the user setup is as follows:
#define USER_SETUP_INFO "User_Setup"
#define ST7789_DRIVER
#define TFT_SDA_READ
#define TFT_RGB_ORDER TFT_RGB
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
#define TFT_INVERSION_OFF
#define TFT_MISO 19
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS 15
#define TFT_DC 2
#define TFT_RST 4
#define TOUCH_CS 21
#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8

I tried using infinite permutations but without success.
Where did I go wrong ?
Any suggestion is welcome.
I tried using the following programs found in the examples:
TFT_Meters (which works, colors aside)
ESP32_SDcard_jpeg (works but images have negative colors)
Touch_calibrate OK

Ultimately, the colors are all different even if I use #define TFT_INVERSION_OFF or ON or don't use it.
I don't know how to arrange the colors correctly.
Ivano

**Comment by Bodmer** [comment]
Try:
#define TFT_RGB_ORDER TFT_BGR

to reverse red and blue.

**Reply by suntchi2018** [reply]
Hi Bodmer, now run well, many thanks
